src/App.tsx

```diff
@@ -15,36 +16,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  SORT_BY_ALPHABET = 'Sort alphabetically',
+  SORT_BY_LENGTH = 'Sort by length',
+}
+
+export interface Change {
+  sortBy: SortType | null;
+  isReversed: boolean;
+}
+
+const sortByAlphabet = (isReversed: boolean, good1: string, good2: string) =>
+  isReversed ? good2.localeCompare(good1) : good1.localeCompare(good2);
+
+function getSortedGoods(goods: string[], { sortBy, isReversed }: Change) {
+  const updateGoods = [...goods];
+
+  if (!sortBy && isReversed) {
+    return updateGoods.reverse();
+  }
+
+  if (sortBy) {
+    return updateGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.SORT_BY_ALPHABET:
+          return sortByAlphabet(isReversed, good1, good2);
+
+        case SortType.SORT_BY_LENGTH:
+          if (good1.length === good2.length) {
+            return sortByAlphabet(isReversed, good1, good2);
+          }
+
+          return isReversed
+            ? good2.length - good1.length
+            : good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return updateGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedGoods = getSortedGoods(goodsFromServer, { sortBy, isReversed });
+
+  const toggleReverse = () => setIsReversed(prevIsReversed => !prevIsReversed);
+
+  const reset = () => {
+    setSortBy(null);
+    setIsReversed(false);
+  };
+
+  const isActivButton = (
+    stateValue: SortType | null | boolean,
+    activeCondition: SortType | boolean,
+  ) => (activeCondition !== stateValue ? 'is-light' : '');
```

Comment:
The function `isActivButton` has a spelling mistake and should be renamed to `isActiveButton` to correctly represent its functionality. Additionally, the name should start with a verb, such as `checkIfButtonIsActive`, to align with the code style guideline. [CODE STYLE]